### ISSUE TITLE
Incorrect spec rendering

### ISSUE THREAD

**Issue by wojtekmach** [elixir-lang/ex_doc#1506]
This spec:

```elixir
@spec foo(String.t(), keyword()) :: :ok
```

is rendered as:

<img width="294" alt="image" src="https://user-images.githubusercontent.com/76071/153052682-a9ea7846-24dc-4562-8fc1-649c08d10e91.png">

there should be no extra newline.

Interestingly, when we switch `keyword()` to `Keyword.t()`, it is rendered correctly. Same if it is `foo(keyword()) :: ...`.

Full snippet to reproduce:

<details>

    Mix.install(
      [
        {:ex_doc, github: "elixir-lang/ex_doc"}
      ],
      elixir: "~> 1.13"
    )

    {:module, module, beam, _} =
      defmodule Foo do
        @spec foo(String.t(), keyword()) :: :ok
        def foo(_, _), do: :ok
      end

    tmp_dir = Path.join(System.tmp_dir!(), "mix_install_ex_doc")
    beam_path = "#{tmp_dir}/_build/dev/lib/example/ebin/#{module}.beam"
    File.mkdir_p!(Path.dirname(beam_path))
    File.write!(beam_path, beam)

    Hex.start()

    defmodule Example.MixProject do
      use Mix.Project

      def project do
        [
          app: :example,
          version: "1.0.0",
          build_path: "#{unquote(tmp_dir)}/_build",
          lockfile: "#{unquote(tmp_dir)}/mix.lock",
          deps_path: "#{unquote(tmp_dir)}/deps"
        ]
      end
    end

    Mix.Task.run("docs", ~w(--formatter html --main Foo --output #{tmp_dir}/doc --open))

</details>

**Comment by timothyvanderaerden** [comment]
This seems to be caused by `Macro.to_string/1`, in Elixir 1.12 this returns `"foo(String.t(), keyword()) :: :ok"` but for Elixir 1.13 it returns `"foo(\n  String.t(),\n  keyword()\n) :: :ok"`

Elixir 1.12
```elixir
iex(1)> Macro.to_string({:"::", [line: 10],
...(1)>  [
...(1)>    {:foo, [line: 10],
...(1)>     [
...(1)>       {{:., [line: 10], [String, :t]}, [line: 10], []},
...(1)>       {:keyword, [line: 0], []}
...(1)>     ]},
...(1)>    :ok
...(1)>  ]})
"foo(String.t(), keyword()) :: :ok"
````

Elixir 1.13
```elixir
iex(2)> Macro.to_string({:"::", [line: 10],
...(2)>  [
...(2)>    {:foo, [line: 10],
...(2)>     [
...(2)>       {{:., [line: 10], [String, :t]}, [line: 10], []},
...(2)>       {:keyword, [line: 0], []}
...(2)>     ]},
...(2)>    :ok
...(2)>  ]})
"foo(\n  String.t(),\n  keyword()\n) :: :ok"
```

**Comment by josevalim** [comment]
This has been fixed in Elixir v1.13 and main branches!
